omnigibson/sensors/vision_sensor.py

```diff
@@ -215,19 +205,12 @@ def initialize_sensors(self, names):
             names (str or list of str): Name of the raw sensor(s) to initialize.
                 If they are not part of self._RAW_SENSOR_TYPES' keys, we will simply pass over them
         """
-        # Standardize the input and grab the intersection with all possible raw sensors
-        names = set([names]) if isinstance(names, str) else set(names)
-        names = names.intersection(set(self._RAW_SENSOR_TYPES.keys()))
-
-        # Initialize sensors
-        sensors = []
+        names = {names} if isinstance(names, str) else set(names)
         for name in names:
-            sensors.append(lazy.omni.syntheticdata.sensors.create_or_retrieve_sensor(self._viewport.viewport_api, self._RAW_SENSOR_TYPES[name]))
-
-        # Suppress syntheticdata warning here because we know the first render is invalid
-        with suppress_omni_log(channels=["omni.syntheticdata.plugin"]):
-            render()
-        render()    # Extra frame required to prevent access violation error
+            self._add_modality_to_backend(modality=name)
+        render()
+        time.sleep(0.05)
+        render()
```

Comment:
This is directly from Josiah's branch. I should have included his comments: `Needs 2 render steps, with some added time in between to force render buffers to populate`. 

Also, here's the render function fyi: 
```
# Duplicate of simulator's render method, used so that this can be done before simulator is created!
def render():
    """
    Refreshes the Isaac Sim app rendering components including UI elements and view ports..etc.
    """
    set_carb_setting(og.app._carb_settings, "/app/player/playSimulations", False)
    og.app.update()
    set_carb_setting(og.app._carb_settings, "/app/player/playSimulations", True)
```